code-challenges.test.js

```diff
@@ -15,42 +15,137 @@
+      expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
+
+//   ● codedMessage › takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+
+//     ReferenceError: codedMessage is not defined
 
 // b) Create the function that makes the test pass.
 
+//Create a function - codedMessage
+//Parameter - string 
+//Go through string recognizing vowels 
+//Use the replace method to replace lowercase and uppercase vowel with given number. Use g to have the replace go through the string. 
+//Return coded string 
+
+
+const codedMessage = (string) => {
+    return string.replace(/a|A/g, 4).replace(/e|E/g, 3).replace(/i|I/g, 1).replace(/o|O/g, 0)
+}
+
+// PASS  ./code-challenges.test.js
+// codedMessage
+//   ✓ takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0. (2 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+describe("wordReturn", () => {
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+        const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+        const letterA = "a"
+        // Expected output: ["Apple", "Banana", "Orange"]
+        const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+        const letterE = "e"
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+      expect(wordReturn(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+      expect(wordReturn(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+  })
 
+//   ● wordReturn › takes in an array of words and a single letter and returns all the words that contain that particular letter
+
+//     ReferenceError: wordReturn is not defined
 
 // b) Create the function that makes the test pass.
 
+//Create a function - wordReturn 
+//Parameter - array, string 
+//Filter through array to identify if the letter is there
+//Add a logical or operator to include if the letter is uppercase
+//Return array containing the letter
+
+const wordReturn = (array, letter) => {
+   return array.filter(value => { 
+       return value.includes(letter) || value.includes(letter.toUpperCase()) 
+   })
+}
```

Comment:
Awesome work here!  While our example letter inputs are lower cased, it is always good to think of any edge case as a stretch.  What would happen if you got an upper cased letter as the input variable?